File: src/main/java/app/events/EventService.java
package app.events;


import app.core.BaseService;

import java.security.Principal;

public interface EventService extends BaseService<Event, Long> {
    Object findAll(Long size, Long page, String text);

    Object findByOwnerId(Long size, Long page, String text, Long id);

    Event save(Event event, Principal principal);
}

File: src/main/resources/public/js/app/service/eventService.js
app.service('eventService', function ($http) {
    this.list = function (text, size, page) {
        return $http.get('api/events', {
            params: {
                text: text,
                size: size,
                page: page
            }
        });
    };

    this.delete = function (id) {
        return $http.delete('api/events' + id);
    };

    this.save = function (event) {
        return $http.post('api/events', event);
    };

    this.findOne = function (id) {
        return $http.get('api/events/' + id);
    };

    this.findByOwnerId = function (text, size, page, id) {
        return $http.get('api/events/findByOwnerId/' + id, {
                params: {
                    text: text,
                    size: size,
                    page: page
                }
            }
        )
    }
});